src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  BreadcrumbLinkProps,
+  BreadcrumbProps as ChakraBreadcrumbProps,
+  Icon,
+  Text,
+} from "@chakra-ui/react"
+import { BiChevronRight } from "react-icons/bi"
+import { Link as RouterLink } from "react-router-dom"
+
+export interface BreadcrumbItem {
+  title: string
+  url?: string
+}
+
+interface BreadcrumbProps {
+  items: BreadcrumbItem[]
+  maxBreadcrumbsLength?: number
+  spacing?: ChakraBreadcrumbProps["spacing"]
+  textStyle?: BreadcrumbLinkProps["textStyle"]
+  isLinked?: boolean
+  isLastChildUnderlined?: boolean
+}
+
+export const Breadcrumbs = ({
+  items,
+  maxBreadcrumbsLength = 3,
+  spacing = "0.5rem",
+  textStyle = "caption-2",
+  isLinked = true,
+  isLastChildUnderlined = false,
+}: BreadcrumbProps): JSX.Element => {
+  if (items.length === 0) {
+    return <></>
+  }
+
+  return (
+    <Breadcrumb
+      spacing={spacing}
+      separator={
+        <Icon
+          as={BiChevronRight}
+          color="interaction.links.neutral-default"
+          lineHeight="0.5rem"
+          textStyle={textStyle}
+        />
+      }
+      lineHeight="0.5rem"
+    >
+      {items.map(({ title, url }, idx) => {
```

Comment:
chakra might have this inbuilt that we could use! 
https://dev.to/estheragbaje/how-to-truncate-text-using-chakra-ui-4mpj

Reply:
Hmm yes but unfortunately we need to perform this truncation ourselves as the page you linked is for texts, and it is when there is not enough space to display all the text. However, in this case we are showing the first item and the last 2 items, then compressing anything in-between into a single ellipsis.